### PR TITLE
Add IsIPv6UniqueLocal property

### LINKED ISSUE #45148: Add IsIPv6UniqueLocal to IPAddress
## Background and Motivation

The `IPAddress` class contains a property `IsIPv6SiteLocal`, but the IPv6 site local range has been deprecated since 2004.

Since October 2005, a new range has been reserved by IETF to denote "Unique Local Addresses" (ULA) for use within private networks. It would be nice if `IPAddress` also had a property that checks for this range.

See also https://en.wikipedia.org/wiki/Unique_local_address

## Proposed API

```csharp
namespace System.Net
{
    public class IPAddress
    {
        public bool IsIPv6UniqueLocal { get; }
    }
}
```

Where the property could be implemented as:

```csharp
public bool IsIPv6UniqueLocal
    => IsIPv6 && ((_numbers[0] & 0xFE00) == 0xFC00);
```

### PR DESCRIPTION
**Pull request by annchous**
IsIPv6UniqueLocal boolean property was added to IPAddress. 
In September 2004 was deprecated the definition of the IPv6 site local range. As a result, in October 2005 for use in private IPv6 networks was defined the associated term **unique local addresses**.

Fix [#45148](https://github.com/dotnet/runtime/issues/45148)

### CODE REVIEW

File: src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs [hunk 1]
```diff
@@ -485,6 +485,19 @@ public bool IsIPv6Teredo
...
+        ///     Determines if an address is an IPv6 Unique Local address
+        ///   </para>
+        /// </devdoc>
+        public bool IsIPv6UniqueLocal
+        {
+            get
+            {
+                return IsIPv6 && ((_numbers[0] & 0xFE00) == 0xFC00);
+            }
+        }
```
Comment (stephentoub) [inline src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs:499]: Thanks for the PR.  To be complete, this also needs to include the new member in the System.Net.Primitives/ref/System.Net.Primitives.cs reference assembly file, and tests.

File: src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs [hunk 2]
```diff
@@ -485,6 +485,19 @@ public bool IsIPv6Teredo
             }
         }
 
+        /// <devdoc>
+        ///   <para>
+        ///     Determines if an address is an IPv6 Unique Local address
+        ///   </para>
+        /// </devdoc>
```
Comment (stephentoub) [inline src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs:492]:
```suggestion
        /// <summary>Gets whether the address is an IPv6 Unique Local address.</summary>
```

File: src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs [hunk 3]
```diff
@@ -485,6 +485,19 @@ public bool IsIPv6Teredo
...
+        public bool IsIPv6UniqueLocal
+        {
+            get
+            {
+                return IsIPv6 && ((_numbers[0] & 0xFE00) == 0xFC00);
```
Comment (danmoseley) [inline src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs:497]: Might need to ! away nullability on _numbers

File: src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs (hunk 2 above)
Comment (annchous) [reply src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs:492]: Previously created implementations of such properties adhere to the documentation I have provided (for example, `IsIPv6LinkLocal`). Is it exactly necessary to replace this with summary?
If a replacement is necessary, then is it necessary to apply such a style to existing similar properties?

File: src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs (hunk 3 above)
Comment (annchous) [reply src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs:497]:
```suggestion
                return IsIPv6 && ((_numbers[0]! & 0xFE00) == 0xFC00);
```

Like this, sure? Got it!

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=48853) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=48853) before we can accept your contribution.<br/><br/>:x: annchous [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=48853)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=48853) it.</sub>

File: src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs (hunk 2 above)
Comment (stephentoub) [reply src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs]: You don't need to update the existing ones: we'll be taking care of that in the coming months. But yes, please use my suggestion instead. Thanks!

File: src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs [hunk 4]
```diff
@@ -485,6 +485,15 @@ public bool IsIPv6Teredo
             }
         }
 
+        /// <summary>Gets whether the address is an IPv6 Unique Local address.</summary>
+        public bool IsIPv6UniqueLocal
+        {
+            get
+            {
+                return IsIPv6 && ((_numbers[0]! & 0xFE00) == 0xFC00);
```
Comment (stephentoub) [inline src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs:493]:
```suggestion
                return IsIPv6 && ((_numbers![0] & 0xFE00) == 0xFC00);
```

Comment (annchous) [reply src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs:493]: sorry, night coding hurts

File: src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs (hunk 2 above)
Comment (annchous) [reply src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs]: Okay, thank you

**Comment by annchous** [comment]
@stephentoub anything else to fix?)

**Comment by geoffkizer** [comment]
LGTM. Thanks!

**Review by geoffkizer — APPROVED** [review]

**Pull request merged**